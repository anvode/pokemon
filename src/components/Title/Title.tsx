
import * as React from 'react';
import { useSelector } from 'react-redux';

import { RootState } from '../../redux/root-reducer';
            
export interface TitleProps {}
const Title: React.FC<TitleProps> = () => {
    const { title } = useSelector((state: RootState) => state.pokemon);
        
    return <>
        <h1 data-testid="h1" className="app__title bg-light p-3 mb-3">{title}</h1>

    </>;
};
            
export default Title;