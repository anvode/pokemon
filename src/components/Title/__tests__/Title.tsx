import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Title from '../Title';

import Store from '../../../redux/store';

describe('App', () => {
    afterEach(cleanup);
    
    it('renders', () => {
        const { asFragment } = render(<Title name="Title"/>, {
            wrapper: function child({ children }) {return <Provider store={Store}><BrowserRouter>{children}</BrowserRouter></Provider>;}
        });
        expect(asFragment()).toMatchSnapshot();
    });

    it('inserts text in h1', async () => {
            
        const { getByTestId } = render(<Title name="Title"/>, {
            wrapper: function child({ children }) {return <Provider store={Store}><BrowserRouter>{children}</BrowserRouter></Provider>;}
        });

        expect(getByTestId('h1')).toHaveClass('app__title');

        expect(getByTestId('h1')).toHaveTextContent('Title');
        
    });

});