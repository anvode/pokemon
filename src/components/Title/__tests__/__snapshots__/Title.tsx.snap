// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<DocumentFragment>
  <h1
    class="app__title  text-center p-3 mb-3"
    data-testid="h1"
  >
    Title
  </h1>
</DocumentFragment>
`;
